x-pack/plugins/security/server/authentication/api_keys/api_keys.ts

```diff
@@ -335,6 +337,32 @@ export class APIKeys {
     return result;
   }
 
+  /**
+   * Tries to validate an API key.
+   * @param apiKey ApiKey.
+   */
+  async validate(apiKey: Pick<CreateAPIKeyResult, 'id' | 'api_key'>) {
+    if (!this.license.isEnabled()) {
+      return { isValid: false };
+    }
+
+    const fakeRequest = getFakeKibanaRequest(apiKey);
+
+    this.logger.debug(`Trying to validate an API key as current user`);
+
+    let result: SecurityAuthenticateResponse;
+
+    try {
+      result = await this.clusterClient.asScoped(fakeRequest).asCurrentUser.security.authenticate();
+      this.logger.debug(`API key was validated successfully`);
+      return { isValid: true, result };
```

Comment:
**question** which fields do you require from `result`? I'd prefer not to expose everything from the authenticate endpoint. Is it enough to just return the boolean for now, or do you need something else? If you need other pieces of information, can we return only what is absolutely required for your usecase?